File: adt2ext5/get_4op_to_test_temp.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

/*static*/ uint16_t get_4op_to_test_temp (void)
{
  uint16_t result = 0;
  int_least16_t n = mn_environment.curr_pos - 1; // current instrument index

  if ((n >= 0) && (n <= 254) && (songdata.flag_4op != 0))
  {
    if (_4op_flag_column[n] == charmap.four_op_upper_half)
      result = n + 1 + 1 + ((n + 1) << 8);
    else if ((n > 0) && (_4op_flag_column[n] == charmap.four_op_lower_half))
      result = n + 1 + ((n - 1 + 1) << 8);
  }
  return result;
}

File: adt2ext5/pas/get_4op_to_test_temp.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

// Saved from "adt2ext5.pas"

function get_4op_to_test_temp: Word;

var
  result: Word;
  curr_inst: Byte;

begin
  result := 0;
  curr_inst := mn_environment.curr_pos;
  If (curr_inst in [1..255]) and (songdata.flag_4op <> 0) then
    If (_4op_flag_column[curr_inst] = _4op_flag_chr_beg) then
      result := SUCC(curr_inst)+curr_inst SHL 8
    else If (curr_inst > 1) and (_4op_flag_column[curr_inst] = _4op_flag_chr_end) then
           result := curr_inst+PRED(curr_inst) SHL 8;
  get_4op_to_test_temp := result;
end;
